test/xpu_api/utilities/utility/utility.swap/swap_array.pass.cpp

```diff
@@ -0,0 +1,147 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/array>
+#include <oneapi/dpl/utility>
+#include <oneapi/dpl/type_traits>
+
+#include "support/utils.h"
+#include "support/move_only.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct CopyOnly
+{
+    CopyOnly() = default;
+    CopyOnly(CopyOnly const&) noexcept = default;
+    CopyOnly&
+    operator=(CopyOnly const&)
+    {
+        return *this;
+    }
+};
+
+struct NoexceptMoveOnly
+{
+    NoexceptMoveOnly() = default;
+    NoexceptMoveOnly(NoexceptMoveOnly&&) noexcept = default;
+    NoexceptMoveOnly&
+    operator=(NoexceptMoveOnly&&) noexcept
+    {
+        return *this;
+    }
+};
+
+struct NotMoveConstructible
+{
+    NotMoveConstructible&
+    operator=(NotMoveConstructible&&)
+    {
+        return *this;
+    }
+
+    NotMoveConstructible(NotMoveConstructible&&) = delete;
+};
+
+struct NotMoveAssignable
+{
+    NotMoveAssignable(NotMoveAssignable&&) = delete;
+
+    NotMoveAssignable&
+    operator=(NotMoveAssignable&&) = delete;
+};

```

Comment:
Consider moving to common header in `utility.swap` since it is duplicated with the next file.

Reply:
Done